### PR TITLE
Use expressjs-style routes

### PR DESCRIPTION
**Pull request by steve-taylor**

### CODE REVIEW

**Comment by darrennolan** [comment]
Need updated docs. How's this work?

**Comment by steve-taylor** [comment]
> Need updated docs. How's this work?

I'll update README.md shortly.

**Comment by steve-taylor** [comment]
@darrennolan README.md and example.html have been updated.

File: index.js [hunk 1]
```diff
@@ -85,9 +86,34 @@ export default function baconRouter(baseUrl, initialPath, ...routesAndReturns) {
                     Format is <base>, <initialPath>, <route-match>, <route-response-function>, <route-match>...`;
             }
 
-            if (typeof route == 'string') {
-                if (route === currentRoute) {
-                    return routeReturns();
+            if (typeof route === 'string') {
+                const [path, searchAndHash = ''] = currentRoute.split('?', 2);
+                const keys = [];
+                const regexp = pathToRegexp(route, keys);
```
[the remark this answers is not in the document]
Comment (steve-taylor) [reply index.js]: Query params are *implicitly* supported by path-to-regexp, so there is no route syntax relating to query params. Therefore, any `?` in the route syntax has nothing to do with query params.

[the remark this answers is not in the document]
Comment (steve-taylor) [reply index.js]: `currentRoute` is being split into `<path> [ ? <search> [ # <hash> ] ]`. `path-to-regexp` applies only to the path. `?search`, if present, is converted into the `query` object.

```js
const [path, searchAndHash = ''] = currentRoute.split('?', 2);
const [search, hash = ''] = searchAndHash.split('#', 2);
```

[the remark this answers is not in the document]
Comment (grishkin) [reply index.js]: cool, I think I misunderstood what the route var was in this case. Looks good!

File: index.js [hunk 2]
```diff
@@ -80,14 +81,38 @@ export default function baconRouter(baseUrl, initialPath, ...routesAndReturns) {
...
             }
 
-            if (typeof route == 'string') {
-                if (route === currentRoute) {
-                    return routeReturns();
+            if (typeof route === 'string') {
+                const [, path = '', search = '', hash = ''] = /^([^?#]*)(?:\?([^#]*))?(?:#(.*))?$/.exec(currentRoute);
+                const keys = [];
+                const regexp = pathToRegexp(route, keys);
+                const matches = regexp.exec(decodeURIComponent(path));
```
[the remark this answers is not in the document]
Comment (steve-taylor) [reply index.js]: It depends how we want to handle programmer errors vs. expected runtime errors.

**Pull request merged**